pac4j-http/src/main/java/org/pac4j/http/credentials/authenticator/DigestAuthenticator.java

```diff
@@ -0,0 +1,15 @@
+package org.pac4j.http.credentials.authenticator;
+
+import org.pac4j.http.credentials.DigestCredentials;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+
+/**
+ * This interface represents the contract to validate digest credentials.
+ *
+ * @author Mircea Carasel
+ */
+public interface DigestAuthenticator extends Authenticator<DigestCredentials> {
```

Comment:
The `validate` call validates credentials and should throw a `CredentialsException` if they don't match. The creation of the profile (and attachement to the credentials) is optional but recommended as the default `ProfileCreator` retrieves the profile attached to the credentials. The `HttpProfile` is some default profile, a specific profile must be created by the `Authenticator` if the need arises...